pinot-timeseries/pinot-timeseries-spi/src/main/java/org/apache/pinot/tsdb/spi/plan/ScanFilterAndProjectPlanNode.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tsdb.spi.plan;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.pinot.tsdb.spi.AggInfo;
+import org.apache.pinot.tsdb.spi.TimeBuckets;
+import org.apache.pinot.tsdb.spi.TimeSeriesLogicalPlanner;
+import org.apache.pinot.tsdb.spi.operator.BaseTimeSeriesOperator;
+
+
+/**
+ * This would typically be the leaf node of a plan-tree generated by a time-series engine's logical planner. At runtime,
+ * this gets compiled to a Combine Operator.
+ * <b>Note:</b> You don't need to pass the time-filter to the filter expression, since Pinot will automatically compute
+ *   the time filter based on the computed time buckets in {@link TimeSeriesLogicalPlanner}.
+ */
+public class ScanFilterAndProjectPlanNode extends BaseTimeSeriesPlanNode {
```

Comment:
nit: Scan&Project is called `Calc` in calcite. We don't have it in MSQE and could be useful.

Reply:
I am planning to rename it since this is also inaccurate right now. We compile this to a combine operator which also does aggregation, so this name is misleading. I am tracking it here: https://github.com/apache/pinot/issues/13957